query/src/main/scala/filodb/query/exec/rangefn/MiscellaneousFunction.scala

```diff
@@ -0,0 +1,95 @@
+package filodb.query.exec.rangefn
+
+import java.util.regex.{Pattern, PatternSyntaxException}
+
+import akka.japi.Option.Some
+import monix.reactive.Observable
+
+import filodb.core.query.{CustomRangeVectorKey, IteratorBackedRangeVector, RangeVector, RangeVectorKey}
+import filodb.memory.format.ZeroCopyUTF8String
+
+trait MiscellaneousFunction {
+  def validator(): Boolean
+
+  def execute(): Observable[RangeVector]
+}
+
+case class LabelReplaceFunction(source: Observable[RangeVector], funcParams: Seq[Any])
+  extends MiscellaneousFunction {
+
+  val labelIdentifier: String= "[a-zA-Z_][a-zA-Z0-9_:\\-\\.]*"
+
+  private def pattern(): Option[Pattern] = {
```

Comment:
I would make this a val instead of def, have it return an PAttern instead of an Option.
Throw any PatternSyntaxException out so it becomes a validation exception at construction time.
Add any other "require" inline outside of any method.
This way, you can remove the validator method from trait too.

Reply:
Made the change